operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -698,17 +729,56 @@ private void addIncludesSearchParameters(String compartmentMemberType, Multivalu
     
     private void addSearchParameterIfNotExcluded(String compartmentMemberType, String code, 
         ResourceType.Value subResourceType, MultivaluedMap<String, String> searchParameters, 
-        List<String> allowedIncludes) {
+        List<String> allowedIncludes, List<String> extraResources) {
         // Need to make sure the search parameter has not been excluded
         String parameterName = compartmentMemberType + ":" + code + ":" + subResourceType.value();
         String simplifiedParameterName = compartmentMemberType + ":" + code;
-        if (allowedIncludes == null || allowedIncludes.contains(parameterName)
-                || allowedIncludes.contains(simplifiedParameterName)) {
+        if ((extraResources != null && extraResources.contains(subResourceType.value())) &&
+                (allowedIncludes == null || allowedIncludes.contains(parameterName)
+                || allowedIncludes.contains(simplifiedParameterName))) {
             searchParameters.add("_include", parameterName);
         } else {
             if (LOG.isLoggable(Level.FINE)) {
                 LOG.fine("Filtering out searchParameter because it is not supported by the server config: " + parameterName);
             }
         }
     }
+    
+    private void readsOfAdditionalAssociatedResources(String compartmentMemberType, List<Entry> newEntries,
+        List<Entry> allEntries, List<String> resourceIds, MultivaluedMap<String, String> searchParameters, 
+        FHIRResourceHelpers resourceHelper, List<String> extraResources) throws Exception {
+        if (extraResources == null || extraResources.isEmpty()) {
+            return;
+        }
+        List<Resource> resourceList = new ArrayList<Resource>();
+        for (Entry entry: newEntries) {
+            // Look up entries by reference
+            List<Reference> references = ReferenceFinder.getReferences(entry.getResource());
+            for (Reference reference: references) {
+                ReferenceValue referenceValue = ReferenceUtil.createReferenceValueFrom(reference, compartmentMemberType);
+                if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_RELATIVE)) {
+                    String externalId = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
```

Comment:
As noted in the code.... We're not always guaranteed a String of Patient/1, it could also be a URI et cetra. is the intent to only support LITERAL_RELATIVE?

Reply:
Added a case for LITERAL_ABSOLUTE.

Reply:
I added Javadoc here.